examples/rag/rag_example.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""
+A simple example for conversation between user and
+an agent with RAG capability.
+"""
+import os
+import argparse
+from loguru import logger
+
+from rag_agents import LlamaIndexAgent, LangChainRAGAgent
+import agentscope
+from agentscope.agents import UserAgent
+
+
+def main() -> None:
+    """A conversation demo"""
+
+    agentscope.init(
+        model_configs=[
+            {
+                "model_type": "dashscope_chat",
+                "config_name": "qwen_config",
+                "model_name": "qwen-max",
+                "api_key": f"{os.environ.get('DASHSCOPE_API_KEY')}",
+            },
+            {
+                "model_type": "dashscope_text_embedding",
+                "config_name": "qwen_emb_config",
+                "model_name": "text-embedding-v2",
+                "api_key": f"{os.environ.get('DASHSCOPE_API_KEY')}",
+            },
+        ],
+    )
+
+    if args.module == "llamaindex":
+        AgentClass = LlamaIndexAgent
+    else:
+        AgentClass = LangChainRAGAgent
+    rag_agent = AgentClass(
+        name="Assistant",
+        sys_prompt="You're a helpful assistant. You need to generate"
+        " answers based on the provided context:\n "
+        "Context: \n {retrieved_context}\n ",
```

Comment:
Please note the order of the context and dialogue history in prompt

Reply:
Fixed: new version separates system prompt and use prompt engine to compose the messages